edfio/edf.py

```diff
@@ -996,6 +994,68 @@ def _set_startdate_with_recording(self, recording: Recording) -> None:
         except AnonymizedDateError:
             self._startdate = datetime.date(1985, 1, 1)
 
+    @property
+    def data_record_duration(self) -> float:
+        """Duration of each data record in seconds."""
+        return self._data_record_duration
+
+    def update_data_record_duration(
+        self,
+        data_record_duration: float,
+        method: Literal["strict", "pad", "truncate"] = "strict",
+    ) -> None:
+        """
+        Update the data record duration.
+
+        This operation will fail if the new duration is incompatible with the current
+        sampling frequencies.
+
+        Parameters
+        ----------
+        data_record_duration : float
+            The new data record duration in seconds.
+        method : `{"strict", "pad", "truncate"}`, default: `"strict"`
+            How to handle the case where the new duration does not divide the Edf
+            duration evenly
+
+            - "strict": Raise a ValueError
+            - "pad": Pad the data with zeros to the next compatible duration
+            - "truncate": Truncate the data to the previous compatible duration (might
+              lead to loss of data)
+        """
+        for signal in self.signals:
+            spr = signal.sampling_frequency * data_record_duration
+            if spr % 1:
+                raise ValueError(
+                    f"Cannot set data record duration to {data_record_duration}: Incompatible sampling frequency {signal.sampling_frequency} Hz"
+                )
+        if method == "pad":
+            new_duration = (
+                ceil(self.duration / data_record_duration) * data_record_duration
+            )
+            self._pad_or_truncate_data(new_duration)
+        elif method == "truncate":
+            new_duration = (
+                floor(self.duration / data_record_duration) * data_record_duration
+            )
+            self._pad_or_truncate_data(new_duration)
+        self._data_record_duration = data_record_duration
+        self._set_num_data_records_with_signals(self.signals)
+
+    def _pad_or_truncate_data(self, new_duration: float) -> None:
+        for signal in self.signals:
+            n_samples = round(new_duration * signal.sampling_frequency)
+            diff = n_samples - len(signal._digital)
+            if diff > 0:
+                physical_pad_value = (
+                    signal.physical_min if signal.physical_min > 0 else 0.0
+                )
+                signal._set_data(
+                    np.pad(signal.data, (0, diff), constant_values=physical_pad_value)
+                )
```

Comment:
If the padding value is outside the physical range, this can lead to an overflow in the digital values. I see two ways to handle that: 
- keep the existing range and raise an error if it is exceeded
- modify the physical range to accomodate the padding value

Allowing users to select between those options (e.g., with a parameter `keep_physical_range` as in `EdfSignal.update_data`) would be nice, but probably complicate parameter handling considerably. I'm not sure what the best solution is here, what do you think?

Reply:
Nevermind, I misunderstood the idea behind the determination of the padding value.